public/scripts/extensions/regex/engine.js

```diff
@@ -55,12 +57,53 @@ export function getScriptsByType(scriptType, { allowedOnly } = DEFAULT_GET_REGEX
             const scopedScripts = characters[this_chid]?.data?.extensions?.regex_scripts;
             return Array.isArray(scopedScripts) ? scopedScripts : [];
         }
+        case SCRIPT_TYPES.PRESET: {
+            if (allowedOnly && !extension_settings?.preset_allowed_regex[main_api]?.includes(getPresetName())) {
```

Comment:
This throws if `preset_allowed_regex` is not defined. The one important coalesce is missing :D


```suggestion
            if (allowedOnly && !extension_settings?.preset_allowed_regex?.[main_api]?.includes(getPresetName())) {
```